.github/workflows/python-app.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
Should we use conda instead of pip?  Genuine question - I'm not sure what's best?  (I guess pip is more lightweight?)

Reply:
yea, in the last few years I moved away from 'conda' and just tried to use pip when possible. Potentially pip is simpler and maybe more widely used. Disadvantage of pip is that conda looks after some non-python packages for us too, which can be pretty useful

Reply:
Cool, yeah, let's stick with `pip` for now.  I guess we can always switch to `conda` if we need its skills at installing non-python packages :)